pyccel/ast/core.py

```diff
@@ -2930,29 +2978,39 @@ class Interface(PyccelAstNode):
     is_argument : bool
         True if the interface is used for a function argument.
 
+    is_imported : bool
+        True if the interface is imported from another file.
+
+    syntactic_node : FunctionDef, default: None
+        The syntactic node that is not annotated.
+
     Examples
     --------
     >>> from pyccel.ast.core import Interface, FunctionDef
     >>> f = FunctionDef('F', [], [], [])
     >>> Interface('I', [f])
     """
-    __slots__ = ('_name','_functions','_is_argument')
+    __slots__ = ('_name','_functions','_is_argument', '_is_imported', '_syntactic_node')
     _attribute_nodes = ('_functions',)
 
     def __init__(
         self,
         name,
         functions,
         is_argument = False,
+        is_imported=False,
+        syntactic_node=None,
         ):
 
         if not isinstance(name, str):
             raise TypeError('Expecting an str')
-        if not isinstance(functions, list):
+        if not isinstance(functions, (list, tuple)):
             raise TypeError('Expecting a list')
```

Comment:
According to the docstring `functions` only needs to be iterable, which makes a lot of sense.

Reply:
so should we keep the check as it is or not?

Reply:
E.g.
```python
I = Interface('sum', f for f in methods if f.name.startswith('sum), ...)
```

Reply:
I guess so. The type is specified here: https://docs.python.org/3/library/types.html#types.GeneratorType
The `iterable` method could be updated to allow this

Reply:
> so should we keep the check as it is or not?

As it is above no because "Given our experiments with Python using the -O optimisations, if we want to check these types then we should use an assert instead of an if check." and you should use the `iterable` function.

As it is currently (i.e. no check). Not clear yet. Personally I think it would be better to keep a check but if Yaman would rather we didn't have one at all that's fine with me too as I think the dev would notice quickly as other things won't work properly

Reply:
@said-hadjout We are suggesting to write
```python
assert iterable(functions)
```
but the function `iterable` needs to be updated in order to allow for generators.